// @eve-sama/ngx-signature-pad/esm2022/lib/ngx-signature-pad.module.mjs
import { OverlayModule } from '@angular/cdk/overlay';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxSignaturePadComponent } from './ngx-signature-pad.component';
import * as i0 from "@angular/core";
class NgxSignaturePadModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.2", ngImport: i0, type: NgxSignaturePadModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.0.2", ngImport: i0, type: NgxSignaturePadModule, declarations: [NgxSignaturePadComponent], imports: [CommonModule, OverlayModule], exports: [NgxSignaturePadComponent] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.0.2", ngImport: i0, type: NgxSignaturePadModule, imports: [CommonModule, OverlayModule] });
}
export { NgxSignaturePadModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.2", ngImport: i0, type: NgxSignaturePadModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, OverlayModule],
                    declarations: [NgxSignaturePadComponent],
                    exports: [NgxSignaturePadComponent]
                }]
        }] });
